common/src/java/org/apache/hadoop/hive/conf/Validator.java

```diff
@@ -357,14 +357,15 @@ public String validate(String value) {
       final Path path = FileSystems.getDefault().getPath(value);
       if (path == null && value != null) {
         return String.format("Path '%s' provided could not be located.", value);
-      }
-      final boolean isDir = Files.isDirectory(path);
-      final boolean isWritable = Files.isWritable(path);
-      if (!isDir) {
-        return String.format("Path '%s' provided is not a directory.", value);
-      }
-      if (!isWritable) {
-        return String.format("Path '%s' provided is not writable.", value);
+      } else if (path != null) {
+        final boolean isDir = Files.isDirectory(path);
+        final boolean isWritable = Files.isWritable(path);
+        if (!isDir) {
+          return String.format("Path '%s' provided is not a directory.", value);
+        }
+        if (!isWritable) {
+          return String.format("Path '%s' provided is not writable.", value);
+        }
       }
       return null;
```

Comment:
I think this should be something else than `null` - did the old behaviour accept `null` as a valid path? I believe `Files.isDirectory(null)` returned false....

looking from the other end: I don't think `null` should be accepted as a "writeabledirectory"

Reply:
Refactored the code to return early when the argument is actually null, the following logic is now simplified to Null and non null path